src/transaction/components/TransactionProvider.tsx

```diff
@@ -73,17 +84,48 @@ export function TransactionProvider({
     hash: writeContractTransactionHash || transactionHash,
   });
 
+  const getTransactionReceipts = useCallback(async () => {
+    const receipts = [];
+    for (const hash of transactionHashArray) {
+      try {
+        const txnReceipt = await waitForTransactionReceipt(config, {
+          hash,
+          chainId,
+        });
+        receipts.push(txnReceipt);
+      } catch (err) {
+        console.error('getTransactionReceiptsError', err);
+        setErrorMessage(genericErrorMessage);
+      }
+    }
+    setReceiptArray(receipts);
+  }, [transactionHashArray]);
+
+  useEffect(() => {
+    if (transactionHashArray.length === contracts.length && contracts?.length > 1) {
```

Comment:
why contracts.length > 1?

Reply:
if a user only passes one contract and is using an EOA i felt like it was better to return just the one receipt in the onSuccess instead of an array with that one receipt inside of it.